### ISSUE TITLE
Android?

### ISSUE THREAD

**Issue by melissanoelle** [Michaelvilleneuve/react-native-document-scanner#1]
Hi there, just wondering if you have any plans on implementing this on Android as well. If not, is the problem straight-forward? Is it something that someone else could tackle? I've found one other repo that claims to do Android document scanning, but it's a WIP still and we haven't been able to get it to run.

Thanks in advance,

M

**Comment by Michaelvilleneuve** [comment]
Hi, 
I wouldn't be able to develop the solution from scratch myself, at least not in Java ! 
But if you find a repo of native Android code that does that, I would be glad to make the react native bridge :).

**Comment by axelkennedal** [comment]
I'd also be very interested in an Android implementation 😄 !

**Comment by ducpt2** [comment]
Hey @Michaelvilleneuve , 
Can you take a look at this lib: https://github.com/beast/react-native-scan-doc, i think it so perfect if your lib can work on Android too. :)
So btw, thank you for your effort. :)

**Comment by Michaelvilleneuve** [comment]
Hi, thank you for the infos ! I've seen these library already, I am not working on this project right now but I might be working on it again soon !

**Comment by ducpt2** [comment]
Hi @Michaelvilleneuve  , I see this project. It work exactly like your, but not have auto capture feature. 
https://github.com/Aniruddha-Tapas/Document-Scanner
I trying to bridge it but it too hard for me :) Please take a look, thanks @Michaelvilleneuve.

**Comment by marcorm** [comment]
I'm very interested, too!!

**Comment by themakerman** [comment]
I wish someone can implement Android version. Don't Wana go into license stuff like scanbot sdks....

**Comment by julianklumpers** [comment]
I would really like to see the android version aswell. good work on the ios !

**Comment by zedtux** [comment]
Any plan to add soon the Android support @Michaelvilleneuve ?

**Comment by Michaelvilleneuve** [comment]
Yes you can revert them, especially If the linking adds any compilation issues

**Comment by zedtux** [comment]
It did. Thank you.

**Comment by Fercell** [comment]
Did any of you have found a working solution for Android?

**Comment by zedtux** [comment]
Nope, still looking for it.

**Comment by veb** [comment]
Here's a basic version for Android: https://bitbucket.org/avwave/rn-doc-scanner-android/src/master/

**Comment by julianklumpers** [comment]
hi @veb 

Thank you for linking this basic library for Android. there is a bit of lack on documentation on how to use this in React Native. I figured out how to get this working.


Here is a basic example:

``` javascript
import React from 'react'
import { View, Button, Image } from 'react-native'

import { RNDocScanner } from 'rn-doc-scanner'

export default extends React.Component {


  state = {
    testImage: null,
  }

  _handleCamera = () => {
    RNDocScanner.getDocumentCrop(false)
      .then(res => {
        console.log(res)
        this.setState({ testImage: res })
      })
      .catch(err => {
        console.log(err)
      })
  }

  render() {
    return (
      <View style={{ flex: 1, backgroundColor: '#fff' }}>
        <Button title="Start Camera" onPress={this._handleCamera} />
        {this.state.testImage && (
          <Image source={{ uri: this.state.testImage }} resizeMode="contain" style={{flex: 1, width: null, height: null}} />
        )}
      </View>
    )
  }
}

```

**Comment by Michaelvilleneuve** [comment]
@julianklumpers please use the other lib to discuss it..

**Comment by NguyenQuangVinh123** [comment]
Hello, Do you have the version for Android? Thanks alot

**Comment by NguyenQuangVinh123** [comment]
@julianklumpers  Hello do you resolve this problem on android

**Comment by andrejunqueira1** [comment]
Hi guys, i'm developing a module for android, is not ready yet, but already works, if you wanna try i think tha can solve some problems.
[react-native-documentscanner-android](https://www.npmjs.com/package/react-native-documentscanner-android)
thanks !

**Comment by Michaelvilleneuve** [comment]
@andrejunqueira1 awesome, do not hesitate to open a PR whenever you think our works can be merged !

**Comment by zedtux** [comment]
@andrejunqueira1 that sound great ! Why don't you fork this project and add your Android version in this lib as @Michaelvilleneuve suggested?

**Comment by andrejunqueira1** [comment]
Thank you guys, i'll take a time to join our projects !

**Comment by nicolasscard** [comment]
Estimados, 

Alguna novedad respecto a esto? @andrejunqueira1 @Michaelvilleneuve 
Saludos

**Comment by Michaelvilleneuve** [comment]
Hello, can anyone confirm that https://github.com/Michaelvilleneuve/react-native-document-scanner/pull/47
works on both android and ios ?

It is a PR based on @andrejunqueira1 's work

I'll merge it once it's working

**Comment by leandromaciel** [comment]
> Hello, can anyone confirm that #47
> works on both android and ios ?
> 
> It is a PR based on @andrejunqueira1 's work
> 
> I'll merge it once it's working

How can I create a clean install with the #47 version for testing?

**Comment by GaetanB999** [comment]
Hello, does anyone have good news about this ?

**Comment by Michaelvilleneuve** [comment]
@GaetanB999 and everyone feel free to try out the android branch, it is working very well for us here at fidme. Android feels less polish and still lacks a few features but both are definitely useable.

Also please note that the twin library https://github.com/Michaelvilleneuve/react-native-perspective-image-cropper is now fully compatible with android :)

**Comment by nicolasscard** [comment]
Estimados,

Ya no estoy necesitando la librería.
Gracias por el soporte y la información.

Saludos
